src/librados/CMakeLists.txt

```diff
@@ -8,8 +8,7 @@ add_library(librados_impl STATIC
 # C/C++ API
 add_library(librados ${CEPH_SHARED}
   librados_c.cc
-  librados_cxx.cc
-  $<TARGET_OBJECTS:common_buffer_obj>)
+  librados_cxx.cc)
```

Comment:
this partially reverts 7bf6b5ee1208a359826c74ab033e6bbbfc65969f and breaks the build of 
```
[ 71%] Linking CXX executable ../../../bin/ceph_test_lazy_omap_stats
/usr/bin/ld: CMakeFiles/ceph_test_lazy_omap_stats.dir/main.cc.o: undefined reference to symbol '_ZN4ceph6buffer7v15_2_03ptr7releaseEv'
//home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/16.0.0-3137-gbc11473b5bc/rpm/el8/BUILD/ceph-16.0.0-3137-gbc11473b5bc/build/lib/libceph-common.so.2: error adding symbols: DSO missing from command line
```

as the stuff in buffer.h are exposed as a public interface of the C++ binding of librados, we need to find a way to expose them. how about defining a function for linking libraries with `-Wl,--version-script=librados.map` and apply it to `ceph-common`? not sure if it'd work though. as we link librados against ceph-common privately on purpose, as most of the stuff in `ceph-common` are not supposed to be visible by librados clients.

Reply:
How about delegating the responsibility to link with `libceph-common` to users of `include/rados/librados.hpp`?

Reply:
that would be that last thing to do... as librados is supposed to be be self-contained from its client's perspective. otherwise, 

1. it breaks backward compatibility. 
1. we need to provide either a `.pc` or a `rados-config.cmake` / `radosConfig.cmake` to encapsulate this linkage in them, and ask user to use pkgconfig or cmake for using librados.

Reply:
These two points are true and valid. Still, didn't they affect **solely** C++ API users of `librados` we don't really care too much?

Reply:
very true. i kept forgetting that! but that change requires us to promote libceph-common.so to a public library, namely to move it to `/usr/lib` so somewhere `ld` can find.